src/k2/cpo/service/Service.cpp

```diff
@@ -132,21 +152,49 @@ seastar::future<> CPOService::start() {
     return seastar::make_ready_future<>();
 }
 
-int makeRangePartitionMap(dto::Collection& collection, const std::vector<String>& eps, const std::vector<String>& rangeEnds) {
-    String lastEnd = "";
+seastar::future<> CPOService::_getNodes() {
+    return AppBase().getDist<HealthMonitor>().invoke_on(_heartbeatMonitorShardId(), &HealthMonitor::getNodepoolEndpoints)
+    .then([this] (std::vector<String>&& nodes) {
+        for (const String& node : nodes) {
+            if (_nodesToCollection.find(node) == _nodesToCollection.end()) {
+                _nodesToCollection[node] = NodeAssignmentEntry{.collection = "", .assigned = false};
+            }
+        }
 
-    if (rangeEnds.size() != eps.size()) {
-        K2LOG_W(log::cposvr, "Error in client's make collection request: collection endpoints size does not equal rangeEnds size");
-        return -1;
+        return seastar::make_ready_future<>();
+    });
+}
+
+String CPOService::_assignToFreeNode(String collection) {
+    auto it = _nodesToCollection.begin();
```

Comment:
I think we should be writing _nodeToCollection to disk somewhere here, and also load it from disk on start.  I wonder if we should also start thinking about our persistence story in the CPO, at least add the initial interface

Reply:
Add TODO comment for CPO persistence with "fileutil::writeFile"